### PR TITLE
[dvc] Collect transformer latency in ingestion stats

### PR DESCRIPTION
**Pull request by elijahgrimaldi**
## Summary

We want to be able to track the latency when using the DaVinci record transformer feature. 

## How was this PR tested?

CI

## Does this PR introduce any user-facing changes?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: clients/da-vinci-client/src/main/java/com/linkedin/davinci/stats/IngestionStats.java [hunk 1]
```diff
@@ -168,6 +170,7 @@ public IngestionStats(VeniceServerConfig serverConfig) {
         localMetricRepository,
         METRIC_CONFIG,
         NEARLINE_LOCAL_BROKER_TO_READY_TO_SERVE_LATENCY);
+    transformerLatencySensor = new WritePathLatencySensor(localMetricRepository, METRIC_CONFIG, TRANSFORMER_LATENCY);
```
Comment (ZacAttack) [inline clients/da-vinci-client/src/main/java/com/linkedin/davinci/stats/IngestionStats.java:173]: We need to register this sensor.  But ideally, we should only register this sensor if the transformer is passed and is being updated.

Let's wrap the register in a function and then invoke that function in the ingestion task if a transformer is passed.

Comment (elijahgrimaldi) [reply clients/da-vinci-client/src/main/java/com/linkedin/davinci/stats/IngestionStats.java]: @ZacAttack Do you mind re-running the tests, I think the integration test A is being flaky

Comment (elijahgrimaldi) [reply clients/da-vinci-client/src/main/java/com/linkedin/davinci/stats/IngestionStats.java]: And also let me know if this looks good to you

**Comment by ZacAttack** [comment]
The failing test suite is stuck on an NPE, let's take a look at that, but otherwise this is looking good.

**Comment by elijahgrimaldi** [comment]
> The failing test suite is stuck on an NPE, let's take a look at that, but otherwise this is looking good.

@ZacAttack Ya you're right, in the testBatchStore it looks like when testing with a record transformer it isn't properly initializing the latency sensor. I will look into it later today.

**Review by ZacAttack — APPROVED** [review]
Looks good!

**Pull request merged**